test/correctness/bitwise_ops.cpp

```diff
@@ -129,22 +129,142 @@ int main(int argc, char **argv) {
         shift_amount = std::min(31, shift_amount);
         int correct = lhs >> shift_amount;
         if (im9(x) != correct) {
-            printf("%x >> (uint8_t)%x -> %x instead of %x\n",
-                   input(x), input(x+1), im9(x), correct);
+            printf("%d >> %d -> %d instead of %d\n",
+                   input(x), shift_amount, im9(x), correct);
             return -1;
         }
     }
 
-    // bitwise and on mixed types
+    // arithmetic shift left with signed type (positive value)
     Func f10;
+    Expr a10 = reinterpret<int>(input(x));
```

Comment:
I think this should just be a cast<int32_t>, rather than a reinterpret (and below). reinterpret is more like bitcast/reinterpret_cast, which I think is not what you want.